src/test/java/com/github/thorbenkuck/netcom2/network/shared/SessionImplTest.java

```diff
@@ -1,66 +1,356 @@
+		session.setProperties(newProperties);
+
+		// Assert
+		assertEquals(newProperties, session.getProperties());
+		assertSame(newProperties, session.getProperties());
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void setPropertiesNull() throws Exception {
+		// Arrange
+		SessionImpl session = new SessionImpl(bridge);
+
+		// Act
+		session.setProperties(null);
+
+		// Assert
+		fail();
 	}
 
 	@Test
 	public void send() throws Exception {
+		// Arrange
+		SessionImpl session = new SessionImpl(bridge);
+		TestObject testObject = new TestObject();
+
+		// Act
+		session.send(testObject);
+
+		// Assert
+		verify(bridge).send(eq(testObject));
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void sendNull() throws Exception {
+		// Arrange
+		SessionImpl session = new SessionImpl(bridge);
+
+		// Act
+		session.send(null);
+
+		// Assert
+		fail();
 	}
 
 	@Test
 	public void eventOf() throws Exception {
+		// Arrange
+		SessionImpl session = new SessionImpl(bridge);
+
+		// Act
+		Pipeline<TestObject> pipeline = session.eventOf(TestObject.class);
+		pipeline.addFirst(testObject -> {
+		});
+
+		// Assert
+		assertNotNull(pipeline);
+		assertEquals(pipeline, session.eventOf(TestObject.class));
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void eventOfNull() throws Exception {
+		// Arrange
+		SessionImpl session = new SessionImpl(bridge);
+
+		// Act
+		session.eventOf(null);
+
+		// Assert
+		fail();
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void triggerEventNotRegistered() throws Exception {
+		// Arrange
+		SessionImpl session = new SessionImpl(bridge);
+
+		// Act
+		session.triggerEvent(TestObject.class, new TestObject());
+
+		// Assert
+		fail();
 	}
 
 	@Test
 	public void triggerEvent() throws Exception {
+		// Arrange
+		SessionImpl session = new SessionImpl(bridge);
+		TestObject testObject = new TestObject();
+		AtomicBoolean finished = new AtomicBoolean(false);
+		session.eventOf(TestObject.class).addFirst(o -> {
+			assertEquals(testObject, o);
+			finished.set(true);
+		});
+
+		// Act
+		session.triggerEvent(TestObject.class, testObject);
+
+		// Assert
+		assertTrue(finished.get());
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void triggerEventObjectNull() throws Exception {
+		// Arrange
+		SessionImpl session = new SessionImpl(bridge);
+
+		// Act
+		session.triggerEvent(TestObject.class, null);
+
+		// Assert
+		fail();
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void triggerEventClassNull() throws Exception {
+		// Arrange
+		SessionImpl session = new SessionImpl(bridge);
+
+		// Act
+		session.triggerEvent(null, new TestObject());
+
+		// Assert
+		fail();
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void triggerEventClassAndObjectNull() throws Exception {
+		// Arrange
+		SessionImpl session = new SessionImpl(bridge);
+
+		// Act
+		session.triggerEvent(null, new TestObject());
```

Comment:
There should be another `null` for the object. This test case is currently a duplicate of #triggerEventClassNull

Reply:
Ups..